### PR TITLE
Refactor the edge/vertex options handling, add README and example image

### PR DESCRIPTION
**Pull request by r0ckarong**
Did some more work. Now the big config hashes are not needed anymore. It will just check for each vertex/edge if a property was set to the options and apply that. Adjusted the test. Updated the README with an example too.

@monora Please have a look.

### CODE REVIEW

**Comment by r0ckarong** [comment]
Thinking of it all the work is now already rebased in #89. That should be the only one that needs merging. Sorry for the spam.

**Comment by r0ckarong** [comment]
Just found a bug that an edge is created that shouldn't be there. I need to look into that.

**Comment by r0ckarong** [comment]
Turns out I misunderstood the shorthand for initializing edges in a DirectedGraph. This PR is good to go.

File: lib/rgl/dot.rb [hunk 1]
```diff
@@ -52,55 +50,42 @@ def to_dot_graph(params = {})
...
-                                          val
-                                        end
-        end
-
-          graph << DOT::Node.new(each_vertex_options)
+        each_vertex_options = default_vertex_options
+
+        if @vertex_options && @vertex_options[v]
+          RGL::DOT::NODE_OPTS.each do |opt|
+            if @vertex_options[v].key?(:"#{opt}")
```
Comment (monora) [inline lib/rgl/dot.rb:64]: Could we perhaps use symbols instead of strings in the OPTS constants? That would spare the conversion here. When the user sets options we could allow both string and symbols. Strings would be converted when setting the value.

I am not sure...

**Review by monora — CHANGES_REQUESTED** [review]
OK, that looks good. I tested my example and your README example at my site. All went fine.
I also generated the doc (`rake yard`, which also looks good.

File: lib/rgl/dot.rb (hunk 1 above)
Comment (r0ckarong) [reply lib/rgl/dot.rb:64]: I can't find the mentioned original `dot.rb` file from the rdoc project that the modified version was derived from so I couldn't see if it had a purpose looking like this. I think symbols should be fine since it's just a big array of options that aren't really used anywhere else from what I can tell.

Could you give me an example of how you would imagine a user to pass a string to the method?

Something like?
```
graph.set_vertex_options('a', label = "My Label")
```

Comment (r0ckarong) [reply lib/rgl/dot.rb:64]: I looked briefly into converting the OPTS constants to symbols and it will break all the rdot tests and certainly need a lot more work to figure out properly. Specifically the "pass me a string that contains both the keyword/symbol and value" to take apart could be quite expensive (in terms of edge cases) when you think about labels or other desired node/edge content strings containing `=`. Maybe this proves trivial to someone else, but I would shy away from this right now. The methods are well defined and allow simple usage for setting complex values without much interpretation. To me, that's good enough.

Comment (monora) [reply lib/rgl/dot.rb:64]: Yes, but symbols are fine. I am not sure, if strings were allowed in the previous version. If so, we should allow them. If not, we can stick with symbols at API level.

Internally having symbols in the OPTS arrays would be an optimization. But since we are dealing only with small graphs, this also is not necessary. I leave it up to you, to dig further.

Comment (r0ckarong) [reply lib/rgl/dot.rb:64]: My suggestion would be to merge the current state and I'll look into it when I have more time for it. This could be a separate improvement.

**Review by monora — APPROVED** [review]

**Comment by monora** [comment]
Rebased and squash your commits. Please check 6338b905802.

**Comment by r0ckarong** [comment]
> Rebased and squash your commits. Please check [6338b90](https://github.com/monora/rgl/commit/6338b905802c3e41607d6f29fd18f22e1923aa84).

The README still had the outdated test in it. #95
